src/iperf_time.c

```diff
@@ -31,35 +31,62 @@
 #include "iperf_config.h"
 #include "iperf_time.h"
 
+// Alias retained to avoid changing the entire codebase
+int
+iperf_time_now(struct iperf_time *now)
+{
+    return iperf_time_now_monotonic(now);
+}
+
 #ifdef HAVE_CLOCK_GETTIME
 
 #include <time.h>
 
 int
-iperf_time_now(struct iperf_time *time1)
+clock_gettime_helper(struct iperf_time *now, clockid_t clk_id)
 {
     struct timespec ts;
     int result;
-    result = clock_gettime(CLOCK_MONOTONIC, &ts);
+    result = clock_gettime(clk_id, &ts);
     if (result == 0) {
-        time1->secs = (uint32_t) ts.tv_sec;
-        time1->usecs = (uint32_t) ts.tv_nsec / 1000;
+        now->secs = (uint32_t) ts.tv_sec;
+        now->usecs = (uint32_t) ts.tv_nsec / 1000;
     }
     return result;
 }
 
+int
+iperf_time_now_monotonic(struct iperf_time *now)
+{
+    return clock_gettime_helper(now, CLOCK_MONOTONIC);
+}
+
+int
+iperf_time_now_wallclock(struct iperf_time *now)
+{
+    return clock_gettime_helper(now, CLOCK_REALTIME);
+}
+
 #else
 
 #include <sys/time.h>
 
 int
-iperf_time_now(struct iperf_time *time1)
+iperf_time_now_monotonic(struct iperf_time *now)
+{
+    // Fall back to wallclock, we can't use monotonic clock.
+    return iperf_time_now_wallclock(now);
+}
+
+int
+iperf_time_now_wallclock(struct iperf_time *now)
 {
     struct timeval tv;
     int result;
+    // Returns the non monotonic local wallclock time.
     result = gettimeofday(&tv, NULL);
-    time1->secs = tv.tv_sec;
-    time1->usecs = tv.tv_usec;
+    now->secs = tv.tv_sec;
+    now->usecs = tv.tv_usec;
```

Comment:
You don't cast here, but do it above. The `struct timeval` uses `time_t` and `suseconds_t`. The `struct timespec` from the `clock_gettime` call uses `time_t` and `long`. The `iperf_time` uses plain `int` for both. Let's cast here as well.

Reply:
good catch